### ISSUE TITLE
SVR training never finishes

### ISSUE THREAD

**Issue by Pimenta14** [haifengl/smile#295]
Hello. I'm trying to train a SVR model to predict some values. My training set has 387 values and I want to predict 165 values. I am using array indexes as features for the model training. 

The problem is each time I run the code it enters in an endless loop.
The console shows this:

70 [main] INFO smile.regression.SVR - SVR finishes 387 SMO iterations
94 [main] INFO smile.regression.SVR - SVR finishes 774 SMO iterations
117 [main] INFO smile.regression.SVR - SVR finishes 1161 SMO iterations
131 [main] INFO smile.regression.SVR - SVR finishes 1548 SMO iterations
146 [main] INFO smile.regression.SVR - SVR finishes 1935 SMO iterations
...... and more each interval of miliseconds, always incrementing SMO iterations

I left the machine working on it 16 hours already, without luck. This happens even if I just want to predict 10, or even 1 value, with smaller training sets too (like 20).

**However**, If I normalize the data set and the features values, the algorithm runs just fine with this output:
80 [main] INFO smile.regression.SVR - SVR finishes training
82 [main] INFO smile.regression.SVR - 176 support vectors, 171 bounded
Success
(this was for the initial size of samples: 387 values of training data / predicting the future 165 values)

I'm doing the normalization with min max formula, and in the end I'm denormalizing it so I can store the "real" values.

Is this a normal behaviour of the SVR training? Do I have any chance of getting it working without data normalization? or there is any other way? I don't know if I'm missing the point and SVR doesn't support values with different scales but from 0 to 1...

**Comment by haifengl** [comment]
Array index is useless as a feature. Even if it finishes with normalization, the model is not trustworthy.

**Comment by Pimenta14** [comment]
Ok, thanks for the info, I will use something like sliding window or I will study other possibilities. My data set is not very rich in terms of features. 

They are, for example, simple sensors (like temperature value of a room) so I only have 2 columns - a timestamp and the returned value, aggregated per hour - That is why I'm using the indexes as features :)

1. But anyway, is it a normal behaviour or not without the normalization? 
2. Even if I use something different as features? 
3. Or this model is not the indicated one for my type of data set?

I guess the point is my 3rd question.

If you can answer I would appreciate :) Thanks again.
